types.cc

```diff
@@ -1555,21 +1555,26 @@ void abstract_type::validate(bytes_view v, cql_serialization_format sf) const {
     visit(*this, validate_visitor{v, sf});
 }
 
-static void serialize_aux(const tuple_type_impl& t, const tuple_type_impl::native_type* v, bytes::iterator& out) {
-    auto do_serialize = [&out] (auto&& t_v) {
-        const data_type& t = boost::get<0>(t_v);
-        const data_value& v = boost::get<1>(t_v);
+static void serialize_aux(const tuple_type_impl& type, const tuple_type_impl::native_type* val, bytes::iterator& out) {
+    assert(val);
+    auto& elems = *val;
+
+    assert(elems.size() <= type.size());
+
```

Comment:
Is illegal to remove fields from UDTs?

Reply:
Yes, you can only add and rename.
https://docs.datastax.com/en/dse/5.1/cql/cql/cql_reference/cql_commands/cqlAlterType.html